src/daft-core/src/series/array_impl/binary_ops.rs

```diff
@@ -232,23 +232,85 @@ impl SeriesBinaryOps for ArrayWrapper<StructArray> {}
 impl SeriesBinaryOps for ArrayWrapper<MapArray> {}
 impl SeriesBinaryOps for ArrayWrapper<ExtensionArray> {}
 impl SeriesBinaryOps for ArrayWrapper<Decimal128Array> {}
-impl SeriesBinaryOps for ArrayWrapper<DateArray> {}
+impl SeriesBinaryOps for ArrayWrapper<DateArray> {
+    fn add(&self, rhs: &Series) -> DaftResult<Series> {
+        use DataType::*;
+        let output_type = (self.data_type() + rhs.data_type())?;
+        let lhs = self.0.clone().into_series();
+        match rhs.data_type() {
+            Duration(tu) => {
+                let intermediate_timestamp_type = Timestamp(*tu, None);
+                let lhs = lhs.cast(&intermediate_timestamp_type)?.as_physical()?;
+                let rhs = rhs.as_physical()?;
+                let physical_result = lhs.add(rhs)?.cast(&intermediate_timestamp_type)?;
+                physical_result.cast(&output_type)
+            }
+            _ => binary_op_unimplemented!(lhs, "+", rhs, output_type),
+        }
+    }
+    fn sub(&self, rhs: &Series) -> DaftResult<Series> {
+        use DataType::*;
+        let output_type = (self.data_type() - rhs.data_type())?;
+        let lhs = self.0.clone().into_series();
+        match rhs.data_type() {
+            Date => {
+                let lhs = lhs.as_physical()?;
+                let rhs = rhs.as_physical()?;
+                let physical_result = lhs.sub(rhs)?;
+                physical_result.cast(&output_type)
```

Comment:
One way we can do this is add a `from_physical` helper constructor to our LogicalArray:

```
DateArray::from_physical(logical_dtype, int32_array)
```

It can just inherit the name and stuff from the physical array